frontend/app/model/Schema.scala

```diff
@@ -0,0 +1,69 @@
+package model
+
+import model.RichEvent.RichEvent
+import play.api.libs.json.Json
+
+object LocationSchema {
+  implicit val writesSchema = Json.writes[LocationSchema]
+}
+
+case class LocationSchema(
+  name: String,
+  address: Option[String],
+  hasMap: Option[String],
+  `@type`: String = "Place"
+)
+
+object OfferSchema {
+  implicit val writesSchema = Json.writes[OfferSchema]
+}
+
+case class OfferSchema(
+  url: String,
+  category: String,
+  price: String,
+  availability: Option[String],
+  `@type`: String = "Offer"
+)
+
+case class EventSchema(
+  name: String,
+  description: Option[String],
+  startDate: String,
+  endDate: String,
+  url: String,
+  image: Option[String],
+  location: Option[LocationSchema],
+  offers: Option[OfferSchema],
+  `@context`: String = "http://schema.org",
+  `@type`: String = "Event"
+)
+
+object EventSchema {
+
+  implicit val writesSchema = Json.writes[EventSchema]
+
+  private def locationOpt(event: RichEvent): Option[LocationSchema] = {
+    event.venue.name.map { name =>
+      LocationSchema(name, event.venue.addressLine, event.venue.googleMapsLink)
+    }
+  }
+
+  // TODO: Generate all offers, not just primary
+  private def offerOpt(event: RichEvent): Option[OfferSchema] = {
+    event.generalReleasePrice.map { price =>
+      OfferSchema(event.memUrl, "primary", price, event.statusSchema)
+    }
+  }
+
+  def from(event: RichEvent): EventSchema = EventSchema(
```

Comment:
This could be an `apply` method (then later you would not need to do `EventSchema.from(event)` but instead `EventSchema(event)`.